code/modules/events/crystal_event.dm

```diff
@@ -467,12 +575,30 @@ This section is for the crystal monsters variations
 	icon_state = "crystal_thug"
 	icon_living = "crystal_thug"
 	icon_dead = "crystal_thug"
-	maxHealth = 50
-	health = 50
-	speed = 1.3
-	harm_intent_damage = 9
-	melee_damage_lower = 15
-	melee_damage_upper = 20
+	maxHealth = 20
+	health = 20
+	speed = 0.9
+	harm_intent_damage = 11
+	melee_damage_lower = 20
+	melee_damage_upper = 35
+	move_force = MOVE_FORCE_NORMAL
+	move_resist = MOVE_FORCE_NORMAL
+	pull_force = MOVE_FORCE_NORMAL
+	dodging = TRUE
+	dodge_prob = 25
+	var/list/temp_turfs
+
+/mob/living/simple_animal/hostile/crystal_monster/thug/attackby(obj/item/O, mob/user, params)
+	for(var/turf/around_turfs in view(7, src))
+		if(!isopenturf(around_turfs) || isspaceturf(around_turfs))
+			continue
+		LAZYADD(temp_turfs, around_turfs)
+	if(prob(30))
+		var/turf/open/choosen_turf = pick(temp_turfs)
+		do_teleport(src, choosen_turf)
+	for(var/i in temp_turfs)
+		LAZYREMOVE(temp_turfs, i)
```

Comment:
Do you need to store `temp_turfs` on the mob?

Reply:
i think? what you mean by that?

Reply:
`var/list/temp_turfs` is defined on the mob object, instead of on the proc
Is it being used anywhere else? Is there need to do this?

Reply:
ohhhh you meant that! no there is no other needs for that, i could move the list in the proc

Reply:
i don't need the LAZYREMOVE if i put it in the proc right?